bridges/PixivBridge.php

```diff
@@ -1,72 +1,109 @@
+		'illustrations/' => 'artworks/',
+		'manga/' => 'artworks/',
+		'novels/' => 'novel/show.php?id='
+	);
+
+	public function collectData() {
+		$content = getContents($this->getSearchURI())
+				 or returnClientError('Unable to query pixiv.net');
+		$content = json_decode($content, true);
+
+		$key = self::JSON_KEY_MAP[$this->getInput('mode')];
 		$count = 0;
-		foreach($content as $result) {
-			if($count == 10) break;
+		foreach($content['body'][$key]['data'] as $result) {
 			$count++;
+			if ($count > $this->getInput('posts')) {
+				break;
+			}
 
 			$item = array();
-			$item['id'] = $result['illustId'];
-			$item['uri'] = 'https://www.pixiv.net/member_illust.php?mode=medium&illust_id=' . $result['illustId'];
-			$item['title'] = $result['illustTitle'];
+			$item['id'] = $result['id'];
+			$item['uri'] = 'https://www.pixiv.net/' . self::WORK_LINK_MAP[$this->getInput('mode')] . $result['id'];
+			$item['title'] = $result['title'];
 			$item['author'] = $result['userName'];
-
-			preg_match_all($timeRegex, $result['url'], $dt, PREG_SET_ORDER, 0);
-			$elementDate = DateTime::createFromFormat('YmdHis',
-						$dt[0][1] . $dt[0][2] . $dt[0][3] . $dt[0][4] . $dt[0][5] . $dt[0][6],
-						new DateTimeZone('Asia/Tokyo'));
-			$item['timestamp'] = $elementDate->getTimestamp();
-
+			$item['timestamp'] = $result['updateDate'];
 			$item['content'] = "<img src='" . $this->cacheImage($result['url'], $item['id']) . "' />";
+
 			$this->items[] = $item;
 		}
 	}
 
+	private function getSearchURI() {
+		$query = urlencode($this->getInput('tag'));
+
+		$uri = static::URI . 'ajax/search/' . $this->getInput('mode')
+			 . $query . '?word=' . $query . '&order=date_d&mode=all&p=1';
+
+		return $uri;
+	}
+
 	private function cacheImage($url, $illustId) {
+		$originalurl = $url;
+
+		if ($this->getInput('fullsize')) {
 
-		$url = str_replace('_master1200', '', $url);
-		$url = str_replace('c/240x240/img-master/', 'img-original/', $url);
+			$url = preg_replace(array(0 => '/pximg\.net\/c\/250x250_80_a2\/.*\/img/m',
+							   1 => '/p0_.*\./m'),
+						 array(0 => 'pximg.net/img-original/img',
+							   1 => 'p0.'),
+						 $url);
+			$illustId .= '_fullsize';
+		}
 		$path = PATH_CACHE . 'pixiv_img/';
 
 		if(!is_dir($path))
 			mkdir($path, 0755, true);
 
-		if(!is_file($path . '/' . $illustId . '.jpeg')) {
-			$headers = array('Referer:  https://www.pixiv.net/member_illust.php?mode=medium&illust_id=' . $illustId);
-			$illust = getContents($url, $headers);
-			if(strpos($illust, '404 Not Found') !== false) {
-				$illust = getContents(str_replace('jpg', 'png', $url), $headers);
+		if(!is_file($path . '/' . $illustId . '.jpg')) {
+			$headers = array('Referer:  https://www.pixiv.net/');
+			try {
+				$illust = getContents($url, $headers);
+			} catch (Exception $e) {
+				$illust = getContents($originalurl, $headers);
 			}
-			file_put_contents($path . '/' . $illustId . '.jpeg', $illust);
+			file_put_contents($path . '/' . $illustId . '.jpg', $illust);
```

Comment:
if $illustId's value is `../../../../etc/rc.local` it could be possible to write outside rss-bridge directory.
You need to escape this situation.